src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -33,21 +32,34 @@ object RockSlicer {
       // Error message will already be printed out by InputProcessor
       System.exit(-1)
     }
-    val (globalOrigin, rockVolume, joints) = inputOpt.get
-    val starterBlocks = Seq(Block(globalOrigin, rockVolume))
+    val (globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs) = inputOpt.get
+    val generatedInput = JointGenerator(globalOrigin, boundingBox, rockVolumeInputs, jointSetInputs)
+    val starterBlocks = Seq(Block(globalOrigin, generatedInput.rockVolume))
 
     // Generate a list of initial blocks before RDD-ifying it
-    val seedBlocks = if (arguments.numProcessors > 1) {
-      val processorJoints = LoadBalancer.generateProcessorJoints(starterBlocks.head, arguments.numProcessors)
-      processorJoints.foldLeft(starterBlocks) { (currentBlocks, joint) =>
-        currentBlocks.flatMap(_.cut(joint))
+    val (seedJoints, remainingJoints) = if (arguments.numProcessors > 1) {
+      // Check if at least one of the input joint sets is persistent
+      if (jointSetInputs.forall(_.persistence != 100.0)) {
+        println("ERROR: Input joint sets must contain at least one persistent joint set to run in parallel. Rerun" +
```

Comment:
Oh yeah, I was forgetting about the age thing too.

Yep, that makes sense. Like I said, I was just curious, but I think I have a better sense now.